### PR TITLE
[DROOLS-4150] make FactHandles serializable

### PR DESCRIPTION
**Pull request by mariofusco**

### CODE REVIEW

**Comment by mariofusco** [comment]
jenkins execute full downstream build

File: drools-core/src/main/java/org/drools/core/common/QueryElementFactHandle.java [hunk 1]
```diff
@@ -97,7 +98,13 @@ public void setObject(Object object) {
         this.object = object;
     }    
 
-    public WorkingMemoryEntryPoint getEntryPoint() {
+    public EntryPointId getEntryPointId() {
+        return null;
+        //throw new UnsupportedOperationException( "DisonnectedFactHandle does not support this method" );
```
Comment (baldimir) [inline drools-core/src/main/java/org/drools/core/common/QueryElementFactHandle.java]: Nitpicking: Please delete this comment line. It is not needed here. Same for the getEntryPoint( InternalWorkingMemory wm ) method.

**Review by baldimir — COMMENTED** [review]
Would it be possible to write a simple test for FactHandle serialization? Or is it already serialized as part of some test?

**Review by baldimir — APPROVED** [review]

File: drools-core/src/main/java/org/drools/core/common/EventFactHandle.java [hunk 2]
```diff
@@ -252,36 +268,39 @@ public void setOtnCount( int otnCount ) {
...
 
     private EventFactHandle cloneWithoutTuples() {
         EventFactHandle clone = new EventFactHandle( getId(),
-                                                     getObject(),
-                                                     getRecency(),
-                                                     getStartTimestamp(),
-                                                     getDuration(),
-                                                     getEntryPoint(),
-                                                     isTraitOrTraitable() );
+                getIdentityHashCode(),
```
Comment (mdproctor) [inline drools-core/src/main/java/org/drools/core/common/EventFactHandle.java:290]: All code should be align on column. not align on indent.

**Pull request merged**